### ISSUE TITLE
MPTimer's timer is not thread safe and causes EXC_BAD_ACCESS

### ISSUE THREAD

**Issue by rob-keepsafe** [mopub/mopub-ios-sdk#318]
One line summary of the issue here

MPTimer's `NSTimer` is nil when init'ed from a background queue and causes an EXC_BAD_ACCESS

<img width="1792" alt="Screen Shot 2020-04-23 at 3 52 58 PM" src="https://user-images.githubusercontent.com/30269720/80166856-ac36d380-857a-11ea-9903-8d914a171cbd.png">

- [x] I am submitting a bug report for existing functionality
- [x] I visited https://developers.mopub.com/ and found no answer
- [x] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [x] I checked to make sure that this issue has not already been filed

#### MoPub SDK Version:

5.12.1

#### Device model and OS Version:

iOS 13.4.1 on iPhone 7

#### Xcode Version:

11.4.1

#### Steps to reproduce the behavior:

Looks to be introduced as a result of https://github.com/mopub/mopub-ios-sdk/commit/3f696fe2f8c6b529fa5e8c0587a98e9f62b9f8ad#diff-4844b334cfea1cf4f07e8e8f04d6456dL43

- Init MoPub SDK from a utility/background queue
- Intermittently hit this race condition when it dispatches back onto the main thread

#### Expected behavior:

The `NSTimer` instance appears to be deallocated since it's not implemented in a thread safe manner

It should likely be reverted to how it was, something like:

```
    void (^mainThreadOperation)(void) = ^void(void) {
        MPTimer *timer = [[MPTimer alloc] init];
        timer.target = target;
        timer.selector = aSelector;
        timer.isCountdownActive = NO;
        timer.isRepeatingTimer = repeats;
        timer.timeInterval = seconds;
        // Initialize the internal `NSTimer`, but set its fire date in the far future.
        // `scheduleNow` will handle the firing of the timer.
        NSTimer *nsTimer = [NSTimer timerWithTimeInterval:seconds
                                                   target:timer
                                                 selector:@selector(timerDidFire)
                                                 userInfo:nil
                                                  repeats:repeats];
        [nsTimer setFireDate:NSDate.distantFuture];
        timer.timer = nsTimer;
        [NSRunLoop.mainRunLoop addTimer:nsTimer forMode:runLoopMode];
    };
```

#### Observed behavior:

Thread safety race condition causes a bad access across threads

#### Evidence:

<img width="1792" alt="Screen Shot 2020-04-23 at 3 52 58 PM" src="https://user-images.githubusercontent.com/30269720/80166856-ac36d380-857a-11ea-9903-8d914a171cbd.png">

**Comment by chauduyphanvu** [comment]
@rob-keepsafe, thanks for letting us know! We do have internal work in progress to resolve this crash. More to come soon!

Vu Chau
MoPub Team

**Comment by chauduyphanvu** [comment]
@rob-keepsafe, this should now be addressed in the 5.13.0 release of the MoPub iOS SDK.
